Mathlib/Tactic/Tauto.lean

```diff
@@ -124,6 +124,7 @@ def distribNot : TacticM Unit := withMainContext do
   liftMetaTactic' (distribNotAux fvars)
 
 /-- Config for the `tauto` tactic. Currently empty. TODO: add `closer` option. -/
+@[nolint structureInType]
```

Comment:
I think structures with no fields should be explicitly excluded from the lint. We basically don't have enough information about how the structure is intended to be used to make an accurate assessment about whether it should be a Prop.

Reply:
that's fair